foreman/data_refinery_foreman/surveyor/sra.py

```diff
@@ -345,220 +205,184 @@ def _apply_metadata_to_experiment(experiment: Experiment, metadata: dict):
             logger.error("Could not discover organism type for run.", accession=run_accession)
             return (None, None)  # This will cascade properly
 
-        organism_name = organism_name.upper()
         organism = Organism.get_object_for_name(organism_name)
 
-        ##
-        # Experiment
-        ##
+        sample_accession = sample_metadata.pop("run_accession")
+        sample, sample_created = Sample.objects.get_or_create(accession_code=sample_accession)
 
-        experiment_accession_code = metadata.get("study_accession")
-        try:
-            experiment_object = Experiment.objects.get(accession_code=experiment_accession_code)
-            logger.debug(
-                "Experiment already exists, skipping object creation.",
-                experiment_accession_code=experiment_accession_code,
-                survey_job=self.survey_job.id,
-            )
-        except Experiment.DoesNotExist:
-            experiment_object = Experiment()
-            experiment_object.accession_code = experiment_accession_code
-            SraSurveyor._apply_metadata_to_experiment(experiment_object, metadata)
-            experiment_object.save()
-
-            ##
-            # Experiment Metadata
-            ##
-            json_xa = ExperimentAnnotation()
-            json_xa.experiment = experiment_object
-            json_xa.data = metadata
-            json_xa.is_ccdl = False
-            json_xa.save()
-
-        ##
-        # Samples
-        ##
-
-        sample_accession_code = metadata.pop("run_accession")
-        # Create the sample object
-        try:
-            sample_object = Sample.objects.get(accession_code=sample_accession_code)
-            # If current experiment includes new protocol information,
-            # merge it into the sample's existing protocol_info.
-            protocol_info, is_updated = self.update_sample_protocol_info(
-                sample_object.protocol_info,
-                experiment_object.protocol_description,
-                experiment_object.source_url,
-            )
-            if is_updated:
-                sample_object.protocol_info = protocol_info
-                sample_object.save()
+        if sample_created:
+            sample.source_database = "SRA"
+            sample.organism = organism
+            sample.manufacturer = sample_metadata.get("instrument_platform", "UNKNOWN")
+            sample.platform_name = sample_metadata.get("instrument_model", "UNKNOWN")
+            sample.platform_accession_code = sample.platform_name.replace(" ", "")
+            sample.technology = "RNA-SEQ"
 
+            # In the rare case that we get something we don't expect
+            # we don't want to process this sample so we can set it to UNKNOWN.
+            if sample.manufacturer not in ["ILLUMINA", "ION_TORRENT"]:
+                sample.manufacturer = "UNKNOWN"
+                sample.platform_name = "UNKNOWN"
+                sample.platform_accession_code = "UNKNOWN"
```

Comment:
@jashapiro the value we want for manufacturer matches exactly what we are getting in `instrument_platform` ie `ILLUMINA` or `ION_TORRENT`. `platform_model` will be the whole name so `Illumina HiSeq 2000` etc. I added back in a check to prevent processing if we do get a weird value back. I checked NextSeq 500 which will return ILLUMINA for platform_name.